cpp/daal/src/threading/threading.h

```diff
@@ -480,10 +617,23 @@ class static_tls
     static_tls_deleter * _deleter    = nullptr;
 };
 
+/// Local storage (LS) for a data of a thread.
+/// Does not change its local variable after nested parallel constructs,
+/// but can have performance penalties comparing to daal::tls.
+/// Can be safely used in case of nested parallel regions.
+///
+/// @tparam F  Type of the data located in the storage
 template <typename F>
 class ls : public tlsBase
 {
 public:
+    /// Initialize a local storage.
+    ///
+    /// @tparam lambdaType  Lambda function of type [/* captures */]() -> F
+    ///
+    /// @param lambda   Lambda function that initializes a local storage
```

Comment:
For some of the comments, you specify that the `@param` is `[in]`, `[out]`, or `[in,out]`. I really like those annotations. Can we have them for all the comments?

```suggestion
    /// @param lambda   Lambda function that initializes local storage
```